controllers/awscluster_controller.go

```diff
@@ -153,9 +152,19 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return reconcile.Result{}, err
 	}
 
+	// Fetch AWSClusterRole from the cluster.
+	awsManagementClusterRoleIdentity := &capa.AWSClusterRoleIdentity{}
+	err = r.Get(ctx, client.ObjectKey{Name: managementAWSCluster.Spec.IdentityRef.Name}, awsManagementClusterRoleIdentity)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
I don't think we should ignore the error when the identity doesn't exist. If we have an `AWSCluster` using a non existing identity, we should return errors.